scalafix/rules/src/main/scala/rsc/rules/pretty/SemanticdbPrinter.scala

```diff
@@ -81,6 +82,43 @@ class SemanticdbPrinter(env: Env, index: DocumentIndex, config: RscCompatConfig)
     case _ => sys.error(s"unsupported tree $tree")
   }
 
+  def symbolOf(signature: Signature): Option[String] =
+    signature match {
+      case TypeSignature(_, lower: TypeRef, upper: TypeRef) if lower == upper => Some(lower.symbol)
+      case _ => None
+    }
+
+
+  def symIsInScope(sym: String, env: Env): Boolean = {
+    extractName(sym)
+      .map(env.lookup) match {
+      // TODO: If the lookup returns NoSymbol, we can insert an import and skip the prefix.
+      // The logic to implement this is left for future work.
+      case None => false
+      case Some(symFromScope) => isSameOrAlias(sym, symFromScope)
+    }
+  }
+
+  def isSameOrAlias(sym: String, envSym: String) =
+    (sym == envSym) || dealias(envSym).contains(sym)
+
+  def dealias(sym: String): Option[String] = {
+    index.symbols.get(sym).flatMap(s => symbolOf(s.signature))
+  }
+
+  def extractName(sym: String): Option[Scala.Names.Name] = {
+    // TODO: At the moment, we return None for local symbols, since they don't have a desc.
+    // The logic to improve on this is left for future work.
+    sym.desc match {
+      case d.Term(value) => Some(n.TermName(value))
+      case d.Type(value) => Some(n.TypeName(value))
+      case d.Package(value) => Some(n.TermName(value))
+      case d.Parameter(value) => Some(n.TermName(value))
+      case d.TypeParameter(value) => Some(n.TypeName(value))
+      case other => None
+    }
+  }
```

Comment:
These helper methods look unrelated to the printer. Please move them into `rsc.rules.semantics`. For example, `extractName` could become an extension method `nameOpt` in `SymbolOps`.